cmd/crc/cmd/console.go

```diff
@@ -31,45 +33,94 @@ var consoleCmd = &cobra.Command{
 	Short:   "Open the OpenShift Web Console in the default browser",
 	Long:    `Open the OpenShift Web Console in the default browser or print its URL or credentials`,
 	Run: func(cmd *cobra.Command, args []string) {
-		if err := runConsole(args); err != nil {
-			exit.WithMessage(1, err.Error())
+		if renderErr := runConsole(os.Stdout, machine.NewClient(), consolePrintURL, consolePrintCredentials, outputFormat); renderErr != nil {
+			exit.WithMessage(1, renderErr.Error())
 		}
 	},
 }
 
-func runConsole(arguments []string) error {
+func showConsole(client machine.Client) (machine.ConsoleResult, error) {
 	consoleConfig := machine.ConsoleConfig{
 		Name: constants.DefaultName,
 	}
 
-	client := machine.NewClient()
 	if err := checkIfMachineMissing(client, consoleConfig.Name); err != nil {
-		return err
+		return machine.ConsoleResult{}, err
 	}
 
-	result, err := client.GetConsoleURL(consoleConfig)
-	if err != nil {
+	return client.GetConsoleURL(consoleConfig)
+}
+func runConsole(writer io.Writer, client machine.Client, consolePrintURL, consolePrintCredentials bool, outputFormat string) error {
+	result, err := showConsole(client)
+	return render(&consoleResult{
+		Success:                 err == nil,
+		State:                   result.State,
+		ClusterConfig:           toConsoleClusterConfig(&result),
+		Error:                   errorMessage(err),
+		ConsolePrintURL:         consolePrintURL,
+		ConsolePrintCredentials: consolePrintCredentials,
+	}, writer, outputFormat)
+}
+
+type consoleResult struct {
+	Success                 bool           `json:"success"`
+	State                   state.State    `json:"state"`
+	Error                   string         `json:"error,omitempty"`
+	ClusterConfig           *clusterConfig `json:"clusterConfig,omitempty"`
+	ConsolePrintURL         bool           `json:"consolePrintUrl"`
```

Comment:
This is an implementation details. I don't think we need it in the json ouput. Just make it private ?

Reply:
sure we can make those private.